src/Sarif/Core/Result.cs

```diff
@@ -73,18 +73,21 @@ public ReportingDescriptor GetRule(Run run = null)
                     return GetRuleByIndex(rules, this.Rule.Index);
                 }
 
-                // Look up by this.RuleDescriptor, Guid, if present
-                if (!string.IsNullOrEmpty(this.Rule?.Guid))
+                // Look up by this.RuleDescriptor.Guid, if present
```

Comment:
>RuleDescriptor [](start = 35, length = 14)

".RuleDescriptor" => ".Rule" (but per Michael, and I agree, we don't need to look up by GUID)

Reply:
Fixed. I'm going to leave it in. Feel free to remove GUID based matching based on how you and Mike resolve the conversation.